### PR TITLE
Di improvements

### PR DESCRIPTION
**Pull request by kafoso**
@Sikian @theevenprime2 

I changed some IoC and dependency injection stuff. Initially, I wanted to use an out-of-the-box library for this, e.g. [`electrolyte`](https://github.com/jaredhanson/electrolyte) or [`sack`](https://github.com/bvalosek/sack). However, these seemed to over-complicate things rather than making them more readable.

I've kept the model structure. However, I've renamed `ConfigModel` to `UtilityModel`, because it now also contains the mvc log.

Also, I renamed most `init` functions to `injectDependencies`.

This structure allows us to write manageable application code as well as unit tests (using mocks).

### CODE REVIEW

**Comment by Sikian** [comment]
I'm starting to be pretty at loss with these changes :P

**Comment by kafoso** [comment]
I summary: We need to inject configurable dependencies from the outside-in rather than the modules fetching the dependencies themselves. This is done so that we can write unit and integration tests.

**Comment by theevenprime2** [comment]
LGTM :+1:

File: server/mvc/models/UtilityModel.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+var UtilityModel = inherit(DefaultModel, {
+  getConfig: function(){
+    return UtilityModel.getConfig();
+  },
+  getLog: function(){
+    return UtilityModel.getLog();
+  },
+}, {
+  getConfig: function(){
+    return config;
```
Comment (kafoso) [inline server/mvc/models/UtilityModel.js]: @Sikian @theevenprime2 
This, for instance, is important. If one of us developers accidentally adds a piece of code that dumps an instance of the `UtilityModel` - which of course is an Object - and it is somehow visible to our users, all the users will see is that `getConfig` is a function. They cannot see what is inside the `config` data. This adds another layer of security.

**Pull request merged**